server/src/repositories/Store/poRepository.ts

```diff
@@ -0,0 +1,187 @@
+      .leftJoin("PurchaseOrderAttachment", "PurchaseOrderAttachment.purchaseOrder", "=", "PurchaseOrder.id")
+      .leftJoin("InventoryItem", "InventoryItem.id", "=", "PurchaseOrderItem.item")
+      .leftJoin("InventoryItemLabel", "InventoryItemLabel.item", "=", "InventoryItem.id")
+      .leftJoin("Label", "Label.id", "=", "InventoryItemLabel.label")
+      .select(
+        "PurchaseOrder.id",
+        "PurchaseOrder.creator",
+        "PurchaseOrder.createDate",
+        "PurchaseOrder.expectedDeliveryDate",
+        "PurchaseOrderItem.id as itemId",
+        "PurchaseOrderItem.item",
+        "InventoryItem.id as invItemId",
+        "InventoryItem.image",
+        "InventoryItem.name",
+        "InventoryItem.unit",
+        "InventoryItem.pluralUnit",
+        "InventoryItem.count",
+        "InventoryItem.pricePerUnit",
+        "Label.label",
+        "PurchaseOrderItem.count as poItemCount",
+        "PurchaseOrderAttachment.id as attachId",
+        "PurchaseOrderAttachment.attachment"
+      );
+    return PurchaseOrderMappper.toDomain(knexResult);
+  }
+
+  public async getPOById(poId: number): Promise<PurchaseOrder> {
+    const knexResult = await this.queryBuilder("PurchaseOrder")
+      .leftJoin("PurchaseOrderItem", "PurchaseOrderItem.purchaseOrder", "=", "PurchaseOrder.id")
+      .leftJoin("PurchaseOrderAttachment", "PurchaseOrderAttachment.purchaseOrder", "=", "PurchaseOrder.id")
+      .leftJoin("InventoryItem", "InventoryItem.id", "=", "PurchaseOrderItem.item")
+      .leftJoin("InventoryItemLabel", "InventoryItemLabel.item", "=", "InventoryItem.id")
+      .leftJoin("Label", "Label.id", "=", "InventoryItemLabel.label")
+      .select(
+        "PurchaseOrder.id",
+        "PurchaseOrder.creator",
+        "PurchaseOrder.createDate",
+        "PurchaseOrder.expectedDeliveryDate",
+        "PurchaseOrderItem.id as itemId",
+        "PurchaseOrderItem.item",
+        "InventoryItem.id as invItemId",
+        "InventoryItem.image",
+        "InventoryItem.name",
+        "InventoryItem.unit",
+        "InventoryItem.pluralUnit",
+        "InventoryItem.count",
+        "InventoryItem.pricePerUnit",
+        "Label.label",
+        "PurchaseOrderItem.count as poItemCount",
+        "PurchaseOrderAttachment.id as attachId",
+        "PurchaseOrderAttachment.attachment"
+      )
+      .where("PurchaseOrder.id", poId);
+    return PurchaseOrderMappper.toDomain(knexResult)[0];
+  }
+
+  public async createNewPO(creatorId: number, expectedDeliveryDate: Date, items: PurchaseOrderItemInput[], attachments: string[]): Promise<PurchaseOrder> {
+    const newId = (await this.queryBuilder("PurchaseOrder")
+    .insert({
+      creator: creatorId,
+      createDate: new Date().toISOString(),
+      expectedDeliveryDate: expectedDeliveryDate,
+    }, "id"))[0];
+
+    if (items.length > 0) {
+     this.addItemsToPO(newId, items);
+    }
+
+    if (attachments.length > 0) {
```

Comment:
Could `items.length > 0` and `attachments.length > 0` be considered business logic that should belong in the service layer instead?

Reply:
eh maybe? Its just a check to see if the repo needs to insert attachments/items. I'll play around with it

Reply:
That's fine, just trying to reach an understanding of what qualifies as business logic.